### PR TITLE
Changes to the main branch

### PR DESCRIPTION
**Pull request by CryceTruly**
I have worked on features of the app and I want to merge these changes to the main repository so that I can update the Heroku application.
Kindly look at the changes and leave some feedback.

### CODE REVIEW

**Comment by walimike** [comment]
@CryceTruly I don't think it's a good idea merging all your work to master as yet, untill your stories have been approved, if it's for the sake of deploying on heroku, then u could as well deploy from another branch, i.e. the one that has all your features implemented

**Comment by walimike** [comment]
Also in the run file, you imported you dev config but did not use it.

**Comment by CryceTruly** [comment]
> Also in the run file, you imported you dev config but did not use it.

Ok, thank you @walimike let me do some refactoring

File: tests/test_users.py [hunk 1]
```diff
@@ -194,3 +184,4 @@ def test_get_a_no_users_message(self):
         if count == 0:
             self.assertEqual(data['msg'], 'No users yet')
         self.assertEqual(response.status, '200 OK')
+
```
Comment (Stanley-Okwii) [inline tests/test_users.py]: Add newline at the end of code

File: app/views/parcels.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+
+# POST /parcels
+@ap.route('/api/v1/parcels', methods=['POST'])
+def add_parcel():
+    '''
+    creates a new parcel order
+    '''
+    if not request.content_type == 'application/json':
+        return jsonify({"failed": 'Content-type must be application/json'}), 415
```
Comment (Stanley-Okwii) [inline app/views/parcels.py]: Error code 415, is used for un supported 'media' type, That means it deals with files,images and others. use 400, instead which stands for a bad request the user is making

Comment (Stanley-Okwii) [reply app/views/parcels.py]: Nice work

File: app/model/parcel.py [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+        if "destination_address" in newparcel and "pickup_address" in newparcel \
+                and "comment_description" in newparcel and "created" in newparcel and \
+                "user_id" in newparcel and "recipient_address" in newparcel and "recipient_phone" in newparcel and \
+                "recipient_email" in newparcel and "status" in newparcel and "recipient_name" in newparcel and "weight" in newparcel:
+            return True
+        else:
+            return False
+
+    def is_order_delivered(self, id):
+        '''
```
Comment (malep2007) [inline app/model/parcel.py]: Maintain a uniform way of documenting your work, if you are using single quotes `'''`, maintain that, if you are using double quotes `"""`, stick to that

File: app/model/parcel.py [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+        checks that we cannot cancel an already delivered order
+        '''
+        for parcel in self.parcels:
+            if parcel['id'] == id:
+                if parcel['status'] == 'delivered':
+                    return True
+        return False
```
Comment (malep2007) [inline app/model/parcel.py]: From what I can see in this logic, this code is supposed to ***check***  a parcel has status ***delivered***, and returns True, but the false is at the wrong indentation level in my opinion. I might be wrong, but please look at this at let me know.

File: tests/test_users.py (hunk 1 above)
Comment (CryceTruly) [reply tests/test_users.py]: noted and working on it

**Comment by CryceTruly** [comment]
Closed due to inactivity

**Pull request closed without merging**